### PR TITLE
Reduce cache usage in workflows

### LINKED ISSUE #3052: [ci] reduce cache usage for some cases
Some workflows need to be optimized to reduce cache usage.

### PR DESCRIPTION
**Pull request by pbchekin**
Fixes #3052.

### CODE REVIEW

File: .github/actions/save/action.yml [hunk 1]
```diff
@@ -2,20 +2,26 @@ name: save
...
   dest:
     description: Directory in cache
     required: true
+  enabled:
+    description: Enable cache
+    default: "true"
 runs:
   using: "composite"
   steps:
     - name: Save ${{ inputs.path }} to cache
```
Comment (anmyachev) [inline .github/actions/save/action.yml:19]: Why not something like that?
```suggestion
    - name: Save ${{ inputs.path }} to cache
      if: ${{ inputs.enabled == 'true' }}
```

Comment (pbchekin) [reply .github/actions/save/action.yml:19]: You are right, it's simpler. I'll make a change.

**Comment by kwasd** [comment]
LGTM. However not sure if pip prone to rate limits.

**Review by kwasd — APPROVED** [review]

**Review by anmyachev — APPROVED** [review]

**Comment by pbchekin** [comment]
> LGTM. However not sure if pip prone to rate limits.

Will re-enable pip cache if we start seeing networking issues.

**Pull request merged**